R/histof.R

```diff
@@ -0,0 +1,231 @@
+#' test for ABO compatibility
+#'
+#' @description ABO compatibility test between donor and candidate
+#' @param cABO A character from 'A', 'B', 'AB', 'O'
+#' @param dABO A character from 'A', 'B', 'AB', 'O'
+#' @param iso A logical value T/F
+#' @return A logical value T/F
+#' @examples
+#' abo(cABO = 'A', dABO = 'A', iso = TRUE)
+#' @export
+abo <- function(cABO = 'A', dABO = 'A', iso = TRUE){
+
+  if(iso == TRUE){
+    value <- cABO == dABO
+  } else {
+    value <- ifelse(dABO == 'O', TRUE,
+                    ifelse(dABO == 'A' & cABO %in% c('A','AB'),TRUE,
+                           ifelse(dABO == 'B' & cABO %in% c('B','AB'),TRUE,
+                                  ifelse(dABO == 'AB' & cABO == 'AB', TRUE, FALSE)
+                                  )
+                           )
+                    )
+  }
+
+  return(value)
+}
+
+#' number of HLA mismatchs
+#'
+#' @description Computes the number of HLA mismatchs between donor and candidate
+#' @param dA donor's HLA-A typing
+#' @param dB donor's HLA-B typing
+#' @param dDR donor's HLA-DR typing
+#' @param cA candidate's HLA-A typing
+#' @param cB candidate's HLA-B typing
+#' @param cDR candidate's HLA-DR typing
+#' @return mmA number of HLA-A mismatchs between \code{dA} and \code{cA};
+#' mmB number of HLA-B mismatchs between \code{dB} and \code{cB};
+#' mmDR number of HLA-DR mismatchs between \code{dA}DRand \code{cDR};
+#' and mmHLA as the sum of mmA + mmB + mmDR
+#' @examples
+#' mmHLA(dA = c('1','2'), dB = c('5','7'), dDR = c('1','4'), cA = c('1','2'), cB = c('03','15'), cDR = c('04','07'))
+#' @export
+mmHLA <- function(dA = c('1','2'), dB = c('5','7'), dDR = c('1','4'),
+                  cA = c('1','2'), cB = c('3','15'), cDR = c('4','7')){
+
```

Comment:
When the function to validate HLA is done, we need to add it here.
Please insert a comment so we don't forget about this.